sway/main.c

```diff
@@ -181,28 +177,8 @@ static void log_kernel() {
 	pclose(f);
 }
 
-static void executable_sanity_check() {
-#ifdef __linux__
-		struct stat sb;
-		char *exe = realpath("/proc/self/exe", NULL);
-		stat(exe, &sb);
-		// We assume that cap_get_file returning NULL implies ENODATA
-		if (sb.st_mode & (S_ISUID|S_ISGID) && cap_get_file(exe)) {
-			wlr_log(WLR_ERROR,
-				"sway executable has both the s(g)uid bit AND file caps set.");
-			wlr_log(WLR_ERROR,
-				"This is strongly discouraged (and completely broken).");
-			wlr_log(WLR_ERROR,
-				"Please clear one of them (either the suid bit, or the file caps).");
-			wlr_log(WLR_ERROR,
-				"If unsure, strip the file caps.");
-			exit(EXIT_FAILURE);
-		}
-		free(exe);
-#endif
-}
 
-static void drop_permissions(bool keep_caps) {
+static void drop_permissions() {
```

Comment:
`drop_permissions(void)`

Reply:
is this a convention here? I neither see other functions in the immediate neighborhood using this, nor CONTRIBUTIN.md mentions it

Reply:
In C, `void call_me_maybe()` declares a function that takes an unspecified number of parameters. So a call like `call_me_maybe(1, 2, 3)` is valid.

To declare a function that takes no parameter, one should use `void call_me_maybe(void)`.

Reply:
@ivyl Leaving it empty is a mistake and should be corrected, along with the other functions in the code.

Reply:
Oh yeah, forgot about that C quirk. Thanks!

2 additional patches to follow this one coming in 5 :-)